src/attestation/Attestation.ts

```diff
@@ -15,12 +15,71 @@ import IRequestForAttestation from '../types/RequestForAttestation'
 import TxStatus from '../blockchain/TxStatus'
 import { factory } from '../config/ConfigLog'
 import Identity from '../identity/Identity'
-import IAttestation from '../types/Attestation'
+import IAttestation, { CompressedAttestation } from '../types/Attestation'
 import { revoke, query, store } from './Attestation.chain'
 import IPublicIdentity from '../types/PublicIdentity'
 
 const log = factory.getLogger('Attestation')
 
+function attestationErrorCheck(attestation: IAttestation): void {
+  if (!attestation.cTypeHash || !attestation.claimHash || !attestation.owner) {
+    throw new Error(
+      `Property Not Provided while building Attestation!\n
+      attestation.cTypeHash:\n
+      ${attestation.cTypeHash}\n
+      attestation.claimHash:\n
+      ${attestation.claimHash}\n
+      attestation.owner:\n
+      ${attestation.owner}`
+    )
+  }
+}
+/**
+ *  Compresses an [[Attestation]] object into an array for storage and/or messaging.
+ *
+ * @param attestation An [[Attestation]] object that will be sorted and stripped for messaging or storage.
+ *
+ * @returns An ordered array of an [[Attestation]].
+ */
+
+export function compressAttestation(
+  attestation: IAttestation
+): CompressedAttestation {
+  attestationErrorCheck(attestation)
+  return [
+    attestation.claimHash,
+    attestation.cTypeHash,
+    attestation.owner,
+    attestation.revoked,
+    attestation.delegationId,
+  ]
+}
+
+/**
+ *  Decompresses an [[Attestation]] from storage and/or message into an object.
+ *
+ * @param attestation A compressesd [[Attestation]] array that is reverted back into an object.
+ *
+ * @returns An object that has the same properties as an [[Attestation]].
+ */
+
+export function decompressAttestation(
+  attestation: CompressedAttestation
+): IAttestation {
+  if (!Array.isArray(attestation) || attestation.length !== 5) {
```

Comment:
I added a simple check to see if the data passed is an `Array` and meets the required length of the `CompressedAttestation`